applications/fishing-map/src/features/workspace/heatmaps/heatmaps.utils.ts

```diff
@@ -1,5 +1,42 @@
+import intersection from 'lodash/intersection'
+import lowerCase from 'lodash/lowerCase'
 import { FISHING_DATASET_TYPE } from 'data/datasets'
 import { UrlDataviewInstance } from 'types'
+import { capitalize } from 'utils/shared'
+
+export const getNotSupportedGearTypesDatasets = (dataview: UrlDataviewInstance) => {
+  const datasetsWithoutGearTypeSupport = dataview?.datasets?.flatMap((dataset) => {
+    const hasGeartype = dataset.schema?.geartype?.enum !== undefined
+    const datasetSelected = dataview.config?.datasets.includes(dataset.id)
+    if (!datasetSelected || hasGeartype) {
+      return []
+    }
+    return dataset
+  })
+  return datasetsWithoutGearTypeSupport
+}
+
+export const getCommonGearTypesInDataview = (dataview: UrlDataviewInstance) => {
+  const activeDatasets = dataview?.datasets?.filter((dataset) =>
+    dataview.config?.datasets.includes(dataset.id)
+  )
+  const geartypes = activeDatasets?.map((d) => d.schema?.geartype?.enum || [])
+  const commonGeartypes = geartypes
+    ? intersection(...geartypes).map((geartype) => ({
+        id: geartype,
+        label: capitalize(lowerCase(geartype)),
```

Comment:
I prefer do the capitalize with css, but maybe is not possible with a label in the selector

Reply:
as the `Select` is a reusable component... do you think does it worth to generalise this behaviour?